contracts/plugins/assets/curve/CurveRecursiveCollateral.sol

```diff
@@ -83,25 +85,42 @@ contract CurveRecursiveCollateral is CurveStableCollateral {
     function refresh() public virtual override {
         CollateralStatus oldStatus = status();
 
-        try this.underlyingRefPerTok() returns (uint192) {
+        try this.underlyingRefPerTok() returns (uint192 underlyingRefPerTok_) {
             // Instead of ensuring the underlyingRefPerTok is up-only, solely check
             // that the pool's virtual price is up-only. Otherwise this collateral
-            // would create default cascades.
+            // would create default cascades when basketsNeeded()/totalSupply() falls.
+
+            // === Check for virtualPrice hard default ===
 
-            // {ref/tok}
+            // {lpToken@t=0/lpToken}
             uint192 virtualPrice = _safeWrap(curvePool.get_virtual_price());
 
-            // {ref/tok} = {ref/tok} * {1}
-            uint192 hiddenReferencePrice = virtualPrice.mul(revenueShowing);
+            // {lpToken@t=0/lpToken}
+            uint192 hiddenVirtualPrice = virtualPrice.mul(revenueShowing);
 
             // uint192(<) is equivalent to Fix.lt
-            if (virtualPrice < exposedReferencePrice) {
-                exposedReferencePrice = virtualPrice;
+            if (virtualPrice < poolVirtualPrice) {
+                poolVirtualPrice = virtualPrice;
                 markStatus(CollateralStatus.DISABLED);
+            } else if (hiddenVirtualPrice > poolVirtualPrice) {
+                poolVirtualPrice = hiddenVirtualPrice;
+            }
+
+            // === Update exposedReferencePrice, ignoring default ===
+
+            // {ref/tok} = {ref/tok} * {1}
+            uint192 hiddenReferencePrice = underlyingRefPerTok_.mul(revenueShowing);
+
+            // uint192(<) is equivalent to Fix.lt
+            if (underlyingRefPerTok_ < exposedReferencePrice) {
+                exposedReferencePrice = underlyingRefPerTok_;
+                // markStatus(CollateralStatus.DISABLED); // don't DISABLE
```

Comment:
If we don't intend to default based on this, why don't we just ignore this whole revenue hiding for the reference price and pass it as-is?

Reply:
Because the core contracts don't want refPerTok() to decrease. If refPerTok() oscillates tightly it can result in seizing RSR, processing revenue, seizing RSR, etc.

Reply:
each time that happens the RToken holders make out like bandits and RSR stakers get rekt

Reply:
refPerTok() should only ever decrease after the inner RToken takes a haircut to basketsNeeded()

Reply:
Oh yeah, this is not the plugin where we're overriding that. Got it, still relies on the exposedReferencePrice for the refPerTok external output.